pkg/net/bitswap.go

```diff
@@ -173,7 +176,10 @@ func (c BitswapClient) SpadeTraversal(parent context.Context,
 	startingCid cid.Cid,
 	maxTraverseDepth uint) (*task.RetrievalResult, error) {
 	logger := logging.Logger("bitswap_client_spade").With("cid", startingCid).With("target", target)
-	cidToRetrieve := startingCid
+
+	numChallenge := validateAndGetNumberOfChallenges(logger)
```

Comment:
rename: `maxChallengesPerLevel`

Also, can we document somewhere about how this variable works -- 

It constrains the number of challenges that will happen per level and crucially will not result in an exponential gain for the number of blocks fetched.

For example if we had `numChallenges` set to 10, that doesn't mean 10 challenges at level 1, 10x10=100 challenges at level 2, 10x100 = 1000 challenges at level 3